### PR TITLE
ffmpeg platform-agnostic hardware-acceleration

### PR DESCRIPTION
**Pull request by phineas-pta**
follow up my comments from #298

ffmpeg can automatically choose the best hardware acceleration (cpu/gpu, intel/apple/amd/nvidia)

also this pull request enables clearing cuda cache after swapping, because on low VRAM + ffmpeg cuda acceleration may produce cuda out-of-memory error

### CODE REVIEW

File: roop/core.py [hunk 1]
```diff
@@ -209,6 +209,7 @@ def start(preview_callback = None):
         process_video_multi_cores(args.source_img, args.frame_paths)
     else:
         process_video(args.source_img, args.frame_paths)
+    torch.cuda.empty_cache()
```
Comment (henryruhs) [inline roop/core.py:212]: You need to wrap this in a condition for nvidia otherwise it gonna kill amd platforms

Comment (phineas-pta) [reply roop/core.py:212]: pytorch doc says it should work on amd platform: https://pytorch.org/docs/stable/notes/hip.html#memory-management

Comment (henryruhs) [reply roop/core.py]: @s0md3v Told me a different story

Comment (phineas-pta) [reply roop/core.py]: i added a gpu check (both amd and nvidia)

Comment (phineas-pta) [reply roop/core.py]: since i don't have amd card, idk why

i changed the file properly

File: roop/core.py [hunk 2]
```diff
@@ -209,6 +209,8 @@ def start(preview_callback = None):
         process_video_multi_cores(args.source_img, args.frame_paths)
     else:
         process_video(args.source_img, args.frame_paths)
+    if args.gpu_vendor = 'nvidia':
```
Comment (henryruhs) [inline roop/core.py:212]: should be `args.gpu_vendor == 'nvidia':`

Comment (phineas-pta) [reply roop/core.py]: sorry typing too quick

**Comment by henryruhs** [comment]
Thanks for the contribution - in case you find more tiny tweaks like that: open another PR

**Comment by phineas-pta** [comment]
no prob, thanks for this great work, i'm still playing around

**Pull request merged**